apiserver/common/block.go

```diff
@@ -33,11 +34,34 @@ type BlockCheckerInterface interface {
 	DestroyAllowed(context.Context) error
 }
 
+// BlockCheckerGetter provides a signature of a function that can be used to
+// late defer getting a [BlockChecker]. This allows block checkers to be made
+// based on the context of the caller.
+type BlockCheckerGetter func(ctx context.Context) (*BlockChecker, error)
+
 // BlockChecker checks for current blocks if any.
 type BlockChecker struct {
 	service BlockCommandService
 }
 
+// DomainServicesGetter describes a type that can be used for getting
+// [services.DomainServices] from a given context that comes from a http
+// request.
+type DomainServicesGetter func(ctx context.Context) (services.DomainServices, error)
+
+// BlockCheckerGetterForServices returns a [BlockCheckerGetter] that is
+// constructed from the supplied context.
+func BlockCheckerGetterForServices(servicesGetter DomainServicesGetter) BlockCheckerGetter {
```

Comment:
This is unused. Instead there's a dupe defined in `apiserver/tools.go`. I would think the one in tools should be deleted and this one used instead.

Reply:
Have deleted the one in common. This was done because the one here would require us mocking out the internal implementation of `BlockChecker`. That is not the interface the handler uses so it gets complicated quickly from a testing perspective.